### PR TITLE
Added a Vagrantfile for easier collaboration

### PR DESCRIPTION
**Pull request by jdbohrman**
Hello, I've tested the build and it's passing on my system so I hope this helps with faster collaboration.

### CODE REVIEW

File: .vagrant/machines/default/virtualbox/vagrant_cwd [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/jdbohrman/Documents/pyswarms
```
Comment (ljvmiranda921) [inline .vagrant/machines/default/virtualbox/vagrant_cwd]: I'm wondering if the hardcoded paths are necessary.

File: .vagrant/machines/default/virtualbox/synced_folders [hunk 2]
```diff
@@ -0,0 +1 @@
+{"virtualbox":{"/vagrant":{"guestpath":"/vagrant","hostpath":"/home/jdbohrman/Documents/pyswarms","disabled":false,"__vagrantfile":true}}}
```
Comment (ljvmiranda921) [inline .vagrant/machines/default/virtualbox/synced_folders]: I'm wondering if the hardcoded paths are necessary.

File: .vagrant/machines/default/virtualbox/vagrant_cwd (hunk 1 above)
Comment (jdbohrman) [reply .vagrant/machines/default/virtualbox/vagrant_cwd]: Hey @ljvmiranda921 great to be here! Here's the info I found on it. The documentation is slim but it appears that is the default insecure private key that Vagrant ships with. Here's a couple links: 

https://www.vagrantup.com/docs/vagrantfile/ssh_settings.html
https://github.com/hashicorp/vagrant/issues/8058

Also, I'm fairly sure you're absolutely correct, as running `vagrant up` would provision the config files and others.

**Comment by ljvmiranda921** [comment]
Note to self: merge to `development` after v.0.3.0 release.

**Pull request merged**